src/utils/Logging.ts

```diff
@@ -44,4 +44,17 @@ if (process.env.NODE_ENV !== "production") {
   );
 }
 
+export async function tryCatchLog<T>(
+  func: () => Promise<T>,
+  logFunction,
+  outputOnFail
+): Promise<T> {
+  try {
+    return await func();
+  } catch (e) {
+    logFunction(`${outputOnFail}: ${e}`);
+  }
+  return null;
+}
+
```

Comment:
Proposing that this could be re-written to work like the following:
```suggestion
export function swallowRejection(
    errorMessage: string,
    errorLog: (message: string) => void = BeastieLogger.error
): (rejection: any) => Promise<null> {
    return async (error: any) => {
        await errorLog(`${errorMessage}: ${JSON.stringify(error)}`);
        return null;
    };
}
```


Then for every promise that you want to "swallow" its possible rejection or error, you just tack on `.catch(swallowRejection("Thing broke"))` to the end of it, like so:

```js
// - mock example:
async function chanceOfRejection(threshold) {
    if (Math.random() > threshold)
        throw 123;
    return 456;
}

// ...
// - so instead of:
const nilOrNumber = await tryCatchLog(() => chanceOfRejection(0.5), "chanceOfRejection error", console.error));
// - you would:
const nilOrNumber = await chanceOfRejection(0.5).catch(swallowRejection("chanceOfRejection error", console.error));
```

If you accept this suggestion, don't forget to update the references. :) If you have questions or thoughts about what I've proposed, feel free to let me know.